weave/tests/test_index.py

```diff
@@ -933,6 +933,63 @@ def test_index_abc_get_parents_complex_fail(test_pantry):
         ind.get_parents(child_path)
 
 
+def test_index_abc_get_parents_max_gen_level(test_pantry):
+    """Test IndexABC get_parents max_gen_level removes uuids where the
+    gen_level is greater than the max_gen_level.
+    """
+
+    # Unpack the test_pantry into two variables for the pantry and index.
+    test_pantry, ind = test_pantry
+
+    # Setup random strucutre of parents and children
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3000")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3003")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_2")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3333")
+
+    tmp_dir = test_pantry.set_up_basket("great_grandparent_3_3")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="3303")
+
+    tmp_dir = test_pantry.set_up_basket("grandparent_2")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="2000", parent_ids=["3000", "3003", "3333"]
+    )
+
+    tmp_dir = test_pantry.set_up_basket("grandparent_2_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="2002")
+
+    tmp_dir = test_pantry.set_up_basket("parent_1")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="1000", parent_ids=["2000", "2002", "3303"]
+    )
+
+    tmp_dir = test_pantry.set_up_basket("parent_1_1")
+    test_pantry.upload_basket(tmp_basket_dir=tmp_dir, uid="1001")
+
+    tmp_dir = test_pantry.set_up_basket("child_0")
+    test_pantry.upload_basket(
+        tmp_basket_dir=tmp_dir, uid="0000", parent_ids=["1001", "1000"]
+    )
+
+    ind.generate_index()
+
+    great_grandparent_uuids = ["3000", "3003", "3333"]
+    parents_df = ind.get_parents("0000", max_gen_level=2)
+
+    parents_df_uuids = parents_df["uuid"].to_list()
+
+    # Find the uuids that are in great_grandparent_uuids and parents_df after
+    # the max_gen_level filter. This should be empty.
+    uuid_diff_list = np.setdiff1d(great_grandparent_uuids,
```

Comment:
I like this function. However, shouldn't the diff list be non-empty? Looking at your comment and the documentation, the return should be everything that is in ggparent uuids that's not in parent uuids... and all of them should not be in the parent one. Can you run this test with a print to make sure it's working like you think?